solr/core/src/java/org/apache/solr/client/solrj/embedded/EmbeddedSolrServer.java

```diff
@@ -71,6 +71,7 @@
   protected final String coreName;
   private final SolrRequestParsers _parser;
   private final RequestWriterSupplier supplier;
+  private boolean mustClearSolrRequestInfo = false;
```

Comment:
This change is highly suspicious.  Why is this boolean now a field of EmbeddedSolrServer?  It doesn't seem like *state* of this object and so I don't think it belongs here.  It seems like a transient status in the course of work within the method that sets/clears the requestInfo.